### PR TITLE
Feature/user information

### PR DESCRIPTION
**Pull request by thebaker10**

### CODE REVIEW

File: src/Components/LoginPage.tsx [hunk 1]
```diff
@@ -26,29 +30,31 @@
...
-            });
+            if(!response.ok) {
+                setAlertMessage(body.data.message);
+                setAlertVisible(true);
+                return;
+            }
+            const user:User = await fetch(env.REACT_APP_API_BASE + '/users/' + body.data.user_id, {credentials: 'include'}).then(b => b.json())
+            $User.next(user.data)
+            const expire = new Date(Date.now()+86400000).toUTCString()
+            document.cookie = `FreshMartUserId=${user.data.userId};Expires=${expire};`;
```
[the remark this answers is not in the document]
Comment (flierc-csp) [reply src/Components/LoginPage.tsx]: We should probably add `Secure;` to the end of the cookie string

File: src/Components/LoginPage.tsx [hunk 2]
```diff
@@ -26,29 +30,31 @@ export function LoginPage() {
...
-
-                setLoading(false);
-                navigate(`/`);
-            });
+            if(!response.ok) {
+                setAlertMessage(body.data.message);
+                setAlertVisible(true);
+                return;
+            }
+            const user:User = await fetch(env.REACT_APP_API_BASE + '/users/' + body.data.user_id, {credentials: 'include'}).then(b => b.json())
```
Comment (noko12312) [inline src/Components/LoginPage.tsx:49]: User cookie is already being created in UserLoginAction.php and UserPostAction.php. I believe it is more efficient to do it there so you do not have to wait for an additional fetch to return the user.

File: src/Pages/Category.tsx [hunk 3]
```diff
@@ -5,12 +5,13 @@ import { SpecialBanner } from "../Components/SpecialBanner";
 import TawkTo from "../Components/TawkTo";
 import React, { useState, useEffect } from 'react';
 import { SliderPlaceholder } from "../Components/Card/SliderPlaceholder";
+import env from "./../env.json"
 
 export function Category() {
     let [categories, setCategories] = useState<any[]>([]);
     let arr:number[] = [1,2,3,4];
     useEffect(() => {
-        fetch(process.env.REACT_APP_API_BASE+"/categories")
```
Comment (noko12312) [inline src/Pages/Category.tsx:13]: Was using process.env not working for you? It has been working for me and Casey. We are using .env.local files to hold our environment variables.

**Review by flierc-csp — APPROVED** [review]

**Review by noko12312 — APPROVED** [review]

File: src/Components/LoginPage.tsx (hunk 2 above)
Comment (flierc-csp) [reply src/Components/LoginPage.tsx:49]: I agree.  We could return the information about the user in the initial response to avoid the additional request.

Comment (thebaker10) [reply src/Components/LoginPage.tsx:49]: I removed the cookie from the client side

https://github.com/thebaker10/Fresh-Mart/pull/44/commits/09b6dce1fd9e5a170d6c9b4654304b7aa2dbc75d

**Pull request merged**